src/test/java/com/algorand/algosdk/integration/Stepdefs.java

```diff
@@ -715,23 +820,32 @@ public void sendMsigTxn() throws JsonProcessingException, ApiException{
     }
 
     @Then("the transaction should go through")
-    public void checkTxn() throws ApiException, InterruptedException{
+    public void checkTxn() throws Exception {
         String ans = acl.pendingTransactionInformation(txid).getFrom();
         assertThat(this.txn.sender.toString()).isEqualTo(ans);
-        acl.waitForBlock(lastRound.add(BigInteger.valueOf(2)));
+        waitForAlgodTransactionProcessingToComplete();
```

Comment:
I believe you may use the pendingTransactionInformation result to give a more precise target to `waitForBlock`. The name of that function is wrong though, it should be `waitForBlockAfter`, so you might need to use `acl.waitForBlock(lastRound)`

Reply:
@winder Resolving with the accompanying explanation - happy to discuss further.

Replacing `waitForAlgodTransactionProcessingToComplete();` with `acl.waitForBlock(lastRound)` appears to work consistently.

However, I prefer to keep as is.  I think `acl.waitForBlock(lastRound)` works because making the algo HTTP request-response duration is long enough for transaction processing to complete.